src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -155,14 +157,24 @@ public void GetDescriptions(DescribeSchemesContext describeContext)
             }
         }
 
+        protected Task<AuthenticationTicket> AuthenticateOnceAsync()
```

Comment:
What is the need for this logic of remembering if _authenticate was set?
What happens if the AuthenticateAsync() returns null?